src/render/tutorials/getting-started-with-docker/2_0.html.md

```diff
@@ -155,72 +188,238 @@ Add in `routingOperations` array following definition:
+
+`HelloWorldHandlerFactory.java`
+```
+package io.knotx.example.hellohandler;
+
+import io.knotx.server.api.handler.RoutingHandlerFactory;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.reactivex.core.Vertx;
+import io.vertx.reactivex.ext.web.RoutingContext;
+
+public class HelloWorldHandlerFactory implements RoutingHandlerFactory {
+
+  @Override
+  public String getName() {
+    return "hellohandler";    // this has to be exactly the same unique string as in operations.conf file                                                 
+  }
+
+  @Override
+  public Handler<RoutingContext> create(Vertx vertx, JsonObject config) {
+    String message = config.getString("message", "Some default message");
+    JsonObject jsonObject = new JsonObject()
+        .put("message", message);
+    return event -> event.response().end(jsonObject.toString());
   }
 }
 ```
-Above file will be processed by [`fragmentsHandler`](https://github.com/Knotx/knotx-fragments/tree/master/handler). 
-We will cover more sophisticated example in the [configurable API gateway tutorial](/tutorials/configurable-api-gateway/) 
-  
+
+The most important thing is to register it by its name. In this case, it was `"hellohandler"`. 
+
+The handling it performs is simply ending the response with previously configured *"Hello World from Knot.x!"* JSON.
+
+Finally, let's update the main project's `settings.gradle.kts` file, so that it will recognize the new module. 
+
+Simply add those two lines:
+```kotlin
+include("hello-module")
+```
+and
+```kotlin
+project(":hello-module").projectDir = file("modules/hello-module")
+```
+
+
+
 ## Docker
 
 ### Configuration
-[Knot.x Starter Kit](https://github.com/Knotx/knotx-starter-kit) project builds docker image. Edit `gradle.properties` and change property `docker.image.name`:
+[Knot.x Starter Kit](https://github.com/Knotx/knotx-starter-kit) project builds either stack (zip distribution) or Docker image. In this tutorial, we'll use the latter.
+ 
+ Let's rename the image. Edit `gradle.properties` and change property `docker.image.name`:
+
+```
+docker.image.name=knotx-example/knotx-docker-tutorial
+```
+
+### Run
 
+Now we are ready to run. First, let's build your docker image:
 ```
-docker.image.name=knotx-example/docker-distribution
+$ gradlew clean build-docker
 ```
 
-## Run
+If you take a closer look at build log (you can use `-i` flag for a more verbose process) you can see, that not only the project is being built, but also it:
+
+- builds the Docker image
+- starts a container for testing
+- runs `healthcheck` test on the container and then executes functional tests (more on that in the next section)
+
+But what *building the Docker image* means in this case?
 
-Now we are ready to run. First, build your docker image
+![Starter Kit Docker Image](/img/blog/getting-started-with-docker/starter-kit-docker.png)
```

Comment:
Is this absolute path to image correct?

Reply:
I was basing on all other existing tutorials and it seems that `/img` is sufficient.